### ISSUE TITLE
ICMP requests are allowed for IPv4 but denied for IPv6

### ISSUE THREAD

**Issue by aanm** [cilium/cilium#3297]
test failed for IPv6 but it was allowed for IPv4
https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/1587/testReport/junit/(root)/runtime/RuntimeValidatedPolicies_L4Policy_Checks/

```
time="2018-03-23T21:12:10Z" level=debug msg="running command: docker exec -i  app1 ping -W 2 -c 5 10.15.202.84"
cmd: "docker exec -i  app1 ping -W 2 -c 5 10.15.202.84" exitCode: 0
 PING 10.15.202.84 (10.15.202.84): 56 data bytes
64 bytes from 10.15.202.84: seq=0 ttl=63 time=0.134 ms
64 bytes from 10.15.202.84: seq=1 ttl=63 time=1.170 ms
64 bytes from 10.15.202.84: seq=2 ttl=63 time=1.152 ms
64 bytes from 10.15.202.84: seq=3 ttl=63 time=0.473 ms
64 bytes from 10.15.202.84: seq=4 ttl=63 time=0.493 ms

--- 10.15.202.84 ping statistics ---
5 packets transmitted, 5 packets received, 0% packet loss
round-trip min/avg/max = 0.134/0.684/1.170 ms

time="2018-03-23T21:12:14Z" level=debug msg="running command: docker exec -i  app1 ping6 -c 5 f00d::a0f:0:0:1150"
cmd: "docker exec -i  app1 ping6 -c 5 f00d::a0f:0:0:1150" exitCode: 1
 PING f00d::a0f:0:0:1150 (f00d::a0f:0:0:1150): 56 data bytes

--- f00d::a0f:0:0:1150 ping statistics ---
5 packets transmitted, 0 packets received, 100% packet loss
```

the policy loaded is under `test/runtime/manifests/Policies-l4-policy.json` and is with the v3 changes.

you can download the PR https://github.com/cilium/cilium/pull/3164, build cilium and import `test/runtime/manifests/Policies-l4-policy.json`. It will be automatically translated to v3. 

the test failing is this one (`pingRequests` is a slice of tests to do `ping` and `ping6`)
```
	It("L4Policy Checks", func() {
		_, err := vm.PolicyImportAndWait(vm.GetFullPath("Policies-l4-policy.json"), helpers.HelperTimeout)
		Expect(err).Should(BeNil())

		for _, app := range []string{helpers.App1, helpers.App2} {
			connectivityTest(pingRequests, app, helpers.Httpd1, true)
			connectivityTest(httpRequestsPublic, app, helpers.Httpd1, true)
			connectivityTest(pingRequests, app, helpers.Httpd2, true)
			connectivityTest(httpRequestsPublic, app, helpers.Httpd2, true)
		}
```

**Comment by joestringer** [comment]
FWIW this requires kernel < 4.11 to reproduce, like the default CI Ginkgo VM kernel. I didn't manage to get to the bottom of why this is the case, but it seems to suggest some issue with the IPv6 CIDR ingress policy datapath code. I double-checked the actual masking operations, which seem to be fine, but the hashmap lookup in the CIDR6 ingress map is not matching the `::/0` CIDR for some reason.

**Comment by joestringer** [comment]
I suspect that #3366 will fix this issue.

**Comment by tgraf** [comment]
Fixed by #3366
